aggregator_api/src/lib.rs

```diff
@@ -136,87 +140,111 @@ async fn post_task<C: Clock>(
-                ),
+                format!("invalid role {role}"),
                 Status::BadRequest,
-            ));
+            ))
         }
-        Vec::from([SecretBytes::new(bytes)])
-    } else {
-        Vec::from([SecretBytes::new(
-            thread_rng()
-                .sample_iter(Standard)
-                .take(req.vdaf.verify_key_length())
-                .collect(),
-        )])
     };
-    let task_expiration = req.task_expiration.map(Time::from_seconds_since_epoch);
-    let time_precision = Duration::from_seconds(req.time_precision);
-    let aggregator_auth_tokens = if let Some(encoded) = req.aggregator_auth_token.as_ref() {
-        let token_bytes = URL_SAFE_NO_PAD.decode(encoded).map_err(|err| {
-            Error::new(
-                format!("Invalid base64 value for aggregator_auth_token: {err}"),
-                Status::BadRequest,
-            )
-        })?;
-        Vec::from([
-            // TODO(#472): Each token in the PostTaskReq should indicate whether it is a bearer
-            // token or a DAP-Auth-Token. For now, assume the latter.
-            DapAuthToken::try_from(token_bytes)
-                .map(AuthenticationToken::DapAuth)
-                .map_err(|_| {
-                    Error::new(
-                        "Invalid HTTP header value in aggregator_auth_token".to_string(),
-                        Status::BadRequest,
-                    )
-                })?,
-        ])
-    } else {
-        // TODO(#472): switch to generating bearer tokens by default
-        Vec::from([AuthenticationToken::DapAuth(random())])
-    };
-    let collector_auth_tokens = match (req.role, req.collector_auth_token) {
-        // TODO(#472): switch to generating bearer tokens by default
-        (Role::Leader, None) => Vec::from([AuthenticationToken::DapAuth(random())]),
-        (Role::Leader, Some(encoded)) => {
-            let token_bytes = URL_SAFE_NO_PAD.decode(encoded).map_err(|err| {
+
+    let vdaf_verify_key_bytes = STANDARD.decode(&req.vdaf_verify_key).map_err(|err| {
+        Error::new(
+            format!("Invalid base64 value for vdaf_verify_key: {err}"),
+            Status::BadRequest,
+        )
+    })?;
+    if vdaf_verify_key_bytes.len() != req.vdaf.verify_key_length() {
+        return Err(Error::new(
+            format!(
+                "Wrong VDAF verify key length, expected {}, got {}",
+                req.vdaf.verify_key_length(),
+                vdaf_verify_key_bytes.len()
+            ),
+            Status::BadRequest,
+        ));
+    }
+
+    // DAP recommends deriving the task ID from the VDAF verify key. We deterministically obtain a
+    // 32 byte task ID by taking SHA-256(VDAF verify key).
+    // https://datatracker.ietf.org/doc/html/draft-ietf-ppm-dap-04#name-verification-key-requiremen
+    let task_id = TaskId::try_from(digest(&SHA256, &vdaf_verify_key_bytes).as_ref())
+        .map_err(|err| Error::new(err.to_string(), Status::BadRequest))?;
```

Comment:
From discussion in Slack: we should incorporate a salt into the task ID derivation, since task ID is public(ish). One option is to have the client of the aggregator API (i.e. divviup-api) pass in the same salt to both aggregators when it provisions the task.

Reply:
After discussing further: the VDAF verify keys we currently deal with for Prio3 and Poplar1 are 16 bytes. The search space of 2^128 possible values is big enough that reversing the SHA-256 hash to get the VDAF verify key from a task ID is already impractical, so adding a salt isn't worth the complexity.